### PR TITLE
Expandable rows

### PR DESCRIPTION
**Pull request by APJohns**

### CODE REVIEW

File: projects/epsilon-blueprint/src/lib/table/table-body/table-body.component.html [hunk 1]
```diff
@@ -1,98 +1,125 @@
...
       <app-actions
-        class="dropdown-menu dropdown-menu-right"
-        attr.aria-labelledby="moreActions{{ rowData[properties.rowId] }}"
+        *ngIf="properties.actions.length < 3; else moreActions"
+        class="action-items"
         [actionItems]="properties.actions"
         [rowId]="properties.rowId"
         [rowData]="rowData"
-        classList="dropdown-item"
+        classList="btn btn-mini"
```
Comment (kaseyllabelle) [inline projects/epsilon-blueprint/src/lib/table/table-body/table-body.component.html:87]: I think we're using btn-sm in the rest of the products now

File: projects/epsilon-blueprint/src/styles/tables.scss [hunk 2]
```diff
@@ -87,12 +87,12 @@
   {
     tr
     {
-      &:nth-child(odd)
+      &tr-stripe-odd
```
Comment (kaseyllabelle) [inline projects/epsilon-blueprint/src/styles/tables.scss:90]: Missing .

File: projects/epsilon-blueprint/src/styles/tables.scss [hunk 3]
```diff
@@ -201,6 +201,16 @@
           vertical-align: middle;
           padding: 0 0 0 15px;
         }
+
+        th.th-expandable
+        {
+          border-right: 0;
```
Comment (kaseyllabelle) [inline projects/epsilon-blueprint/src/styles/tables.scss:207]: Did we need to add a max width here (or somewhere)?

File: projects/epsilon-blueprint/src/styles/tables.scss [hunk 4]
```diff
@@ -212,6 +222,24 @@
         {
           padding: 0 0 0 15px;
         }
+
+        td.td-expandable
+        {
+          padding: 0 0 0 0;
```
Comment (kaseyllabelle) [inline projects/epsilon-blueprint/src/styles/tables.scss:228]: We can use shorthand here

File: projects/epsilon-blueprint/src/styles/tables.scss [hunk 5]
```diff
@@ -212,6 +222,24 @@
...
+
+          button
+          {
+            font-size: $font-size-display2;
+            color: $font-color-primary;
```
Comment (kaseyllabelle) [inline projects/epsilon-blueprint/src/styles/tables.scss:233]: What's the hover/focus styling of this? I think btn-icon-inline is blueberry by default. We might need to add more here.

File: projects/epsilon-blueprint/src/styles/tables.scss [hunk 6]
```diff
@@ -212,6 +222,24 @@
...
+            min-width: 44px;
+            min-height: 44px;
```
Comment (kaseyllabelle) [inline projects/epsilon-blueprint/src/styles/tables.scss:235]: Can we move these above color and move font-size below color to be consistent with outside-in ordering?

File: src/app/modules/component-library/library/table/overview/overview.component.html [hunk 7]
```diff
@@ -26,4 +26,8 @@ <h2>Example</h2>
 </section>
 <ng-template #templateColumn let-rowData>
   Template column example for row {{ rowData.id }}
+</ng-template>
+<ng-template #expandableRowsTemplate let-rowData>
+  <p>Expandable row content example. This is tied to row {{ rowData.id }}.</p>
+  <p>With amount {{ rowData.amount }}</p>
```
Comment (kaseyllabelle) [inline src/app/modules/component-library/library/table/overview/overview.component.html:32]: I wonder if we should clear the bottom margin for the final element in the expandable row?

File: projects/epsilon-blueprint/src/lib/table/table-body/table-body.component.html (hunk 1 above)
Comment (APJohns) [reply projects/epsilon-blueprint/src/lib/table/table-body/table-body.component.html:87]: Should this be addressed in another PR?

File: projects/epsilon-blueprint/src/styles/tables.scss (hunk 5 above)
Comment (APJohns) [reply projects/epsilon-blueprint/src/styles/tables.scss:233]: It stays midnight on hover/focus. Should that change?

File: projects/epsilon-blueprint/src/styles/tables.scss [hunk 8]
```diff
@@ -201,6 +230,11 @@
           vertical-align: middle;
           padding: 0 0 0 15px;
         }
+
+        th.th-expandable + .th-selectable
```
Comment (kaseyllabelle) [inline projects/epsilon-blueprint/src/styles/tables.scss:234]: I think we're hitting the point where we should refactor this stylesheet to group styles related to their features...
```
.bp-table
{
  &.table-sortable { }

  &.table-actions { }

  &.table-expandable-rows { }

  &.table-selectable-rows { }

  &.table-expandable-rows.table-selectable-rows { }

  &.table-compact
  {
    &.table-sortable { }

    &.table-actions { }

    &.table-expandable-rows { }

    &.table-selectable-rows { }

    &.table-expandable-rows.table-selectable-rows { }
  }
}

.table-bordered-rows { }
```

**Pull request merged**